work_queue/src/bindings/python3/work_queue.binding.py

```diff
@@ -821,6 +839,133 @@ def resources_allocated(self):
+        self._create_wrapper()
+
+        self._output_loaded = False
+        self._output = None
+
+        super(PythonTask, self).__init__(self._command)
+        self._specify_IO_files()
+
+    ##
+    # returns the result of a python task as a python variable
+    #
+    # @param self	reference to the current python task object
+    @property
+    def output(self):
+        if not self._output_loaded:
+            if self.result == WORK_QUEUE_RESULT_SUCCESS:
+                try:
+                    with open(os.path.join(self._tmpdir, 'out_{}.p'.format(self._id)), 'rb') as f:
+                        self._output = dill.load(f)
+                except Exception as e:
+                    self._output = e
+            else:
+                self._output = PythonTaskNoResult()
+        return self._output
+
+
+    def __del__(self):
+        try:
+            if self._tmpdir and os.path.exists(self._tmpdir):
+                shutil.rmtree(self._tmpdir)
+
+        except Exception as e:
+            sys.stderr.write('could not delete {}: {}\n'.format(self._tmpdir, e))
+
+
+    ##
+    # Cretes the command to be executed by task. pickles function and arguments.
+    # func 	function to be executed by the task
+    # args	arguments used in function to be executed by task
+    def _python_function_command(self, func, *args):
+        tb = traceback.extract_stack()
+        tb_list = traceback.format_list(tb)
+        caller = tb_list[len(tb_list)-3].split()[1]
+        caller = caller.replace('"', '')
+        caller = caller.replace(',', '')
+
+        with open(self._func_file, 'wb') as wf:
+            dill.dump(func, wf)
+        with open(self._args_file, 'wb') as wf:
+            dill.dump([*args], wf)
+
+        command = './{pprun} -e {tar} --unpack-to "$WORK_QUEUE_SANDBOX"/{unpack}-env python {wrapper} {function} {args} {out}'.format(
+                pprun=os.path.basename(self._pp_run),
+                unpack=os.path.basename(self._tar_file),
+                tar=os.path.basename(self._tar_file),
+                wrapper=os.path.basename(self._wrapper),
+                function=os.path.basename(self._func_file),
+                args=os.path.basename(self._args_file),
+                out=os.path.basename(self._out_file))
+
+
+        return command
+
+
+    def _specify_IO_files(self):
+        self.specify_input_file(self._wrapper, cache=True)
+        self.specify_input_file(self._pp_run, cache=True)
+        self.specify_input_file(self._tar_file, cache=False)
+        self.specify_input_file(self._func_file, cache=False)
+        self.specify_input_file(self._args_file, cache=False)
+        self.specify_output_file(self._out_file, cache=False)
+
+
+    ##
+    # creates the wrapper script which will execute the function. pickles output.
+    def _create_wrapper(self):
+        with open(self._wrapper, 'w') as f:
+            f.write(textwrap.dedent('''\
+                    import sys
+                import dill
```

Comment:
This indent of `import sys` caused an error I believe. After unindented it, it ran and completed successfully.